File: Pragma_Examples/OpenMP/C/6_device_routines/1_device_routine/1_device_routine_usm/compute.c
// Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
// This software is distributed under the MIT License
//
// This example was created by Johanna Potyka

#pragma omp requires unified_shared_memory
#pragma omp declare target
void compute(double *x){
   *x = 1.0;
}
#pragma omp end declare target

File: Pragma_Examples/OpenMP/Fortran/6_device_routines/device_routine_with_interface/4_device_routine_device_type/compute.f90
! Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
! This software is distributed under the MIT License
!
! This example was created by Johanna Potyka
      !--- device routine
      subroutine compute(x)
          implicit none
          !$omp declare target link(compute) device_type(nohost)
          !--------------------
          !example routine called from kernel
          !--- variables
          integer,parameter :: rk=8
          real(kind=rk),intent(inout) :: x
          !x               a value (from array)

          !---
          x = 1.0_rk

      end subroutine compute
